pkg/pull/signature.go

```diff
@@ -88,41 +101,48 @@ func verify(message, signature, publicKeyPEM []byte) error {
 	return nil
 }
 
-func getMessageFromLicense(license *kotsv1beta1.License) ([]byte, error) {
-	// JSON marshaller will sort map keys automatically.
-	fields := map[string]string{
-		"apiVersion":             license.APIVersion,
-		"kind":                   license.Kind,
-		"metadata.name":          license.GetObjectMeta().GetName(),
-		"spec.licenseID":         license.Spec.LicenseID,
-		"spec.appSlug":           license.Spec.AppSlug,
-		"spec.channelName":       license.Spec.ChannelName,
-		"spec.endpoint":          license.Spec.Endpoint,
-		"spec.isAirgapSupported": fmt.Sprintf("%t", license.Spec.IsAirgapSupported),
-	}
-
-	if license.Spec.LicenseSequence > 0 {
-		fields["spec.licenseSequence"] = fmt.Sprintf("%d", license.Spec.LicenseSequence)
-	}
-
-	for k, v := range license.Spec.Entitlements {
-		key := fmt.Sprintf("spec.entitlements.%s", k)
-		val := map[string]string{
-			"title":       v.Title,
-			"description": v.Description,
-			"value":       fmt.Sprintf("%v", v.Value.Value()),
-		}
-		valStr, err := json.Marshal(val)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to marshal entitlement value: %s", k)
-		}
-		fields[key] = string(valStr)
+func verifyLicenseData(outerLicense *kotsv1beta1.License, innerLicense *kotsv1beta1.License) error {
+	if outerLicense.Spec.AppSlug != innerLicense.Spec.AppSlug {
+		return errors.New("\"appSlug\" field has changed")
+	}
+	if outerLicense.Spec.Endpoint != innerLicense.Spec.Endpoint {
```

Comment:
@marccampbell mentioned that he may want to be able to change the endpoint. you should ask him about this

Reply:
perhaps that should be done with an override param elsewhere?
And also have "can override endpoint" be a license param

Reply:
@salahalsaleh1993 I agree with Ethan. The endpoint parameter should be configurable by the end customer here. It’s not part of the product yet, but some end customers could have proxies and want to redirect the endpoint to a proxy or other policy engine to pull. 

Let’s keep endpoint in the license spec, but it should not be validated during the signature.

Reply:
cool, will update, thanks!